### PR TITLE
Incomplete support for ctags/etags generation

### PR DESCRIPTION
**Pull request by jacereda**
This is my attempt at #890. Still lacks ADT constructors and I don't know what else.

To try it with vi:

    psc-docs --format ctags `find bower_components src test -name \*.purs -not -name externs.purs` `find bower_components src test -name externs.purs`  > tags

vi src/Main.purs
position the cursor on top of a symbol and press CTRL+], go back with CTRL+T.

With emacs:

    psc-docs --format etags `find bower_components src test -name \*.purs -not -name externs.purs` `find bower_components src test -name externs.purs`  > TAGS

emacs src/Main.purs
position the cursor on top of a symbol and press ALT+.

### CODE REVIEW

File: psc-docs/Main.hs [hunk 1]
```diff
@@ -251,13 +290,25 @@ inputFile = strArgument $
...
      long "hierarchy-images"
   <> help "Include markdown for type class hierarchy images in the output."
 
+readFormat :: String -> Format
+readFormat "etags" = Etags
+readFormat "ctags" = Ctags
+readFormat _ = Markdown
+
+format :: Parser Format
+format = option (str >>= pure . readFormat) $ value Markdown
```
Comment (paf31) [inline psc-docs/Main.hs]: Maybe use `auto`? 
https://hackage.haskell.org/package/optparse-applicative-0.11.0.2/docs/Options-Applicative-Builder.html#g:3

File: psc-docs/Main.hs [hunk 2]
```diff
@@ -72,6 +84,33 @@ atIndent indent text =
 ticks :: String -> String
 ticks = ("`" ++) . (++ "`")
 
+lineNumber :: P.Declaration -> Int
+lineNumber (P.PositionedDeclaration sp _ _) = P.sourcePosLine $ P.spanStart $ sp
+lineNumber _ = error "Not a PositionedDeclaration"
+
+taggable :: P.Declaration -> Bool
```
Comment (paf31) [inline psc-docs/Main.hs]: Can this be moved inside `taggables` in a `where` clause?

File: psc-docs/Main.hs [hunk 3]
```diff
@@ -31,21 +31,33 @@ import qualified Paths_purescript as Paths
 import System.Exit (exitSuccess, exitFailure)
 import System.IO (hPutStrLn, stderr)
 
+data Format = Markdown
```
Comment (paf31) [inline psc-docs/Main.hs]: Probably worth a line comment for those of us (like me) who don't know what these formats are for :)

**Comment by paf31** [comment]
Looks good generally. I think this might be a good time to break up `Main` into submodules: `Markdown`, `ETags` and `CTags` seems like a sensible division, with `Utils` for your common code.

**Comment by jacereda** [comment]
I still need some hint on how to add the ADT constructors to the generated tags.

**Comment by paf31** [comment]
@jacereda Which constructors are you referring to?

I didn't see your latest commits until just now, but if you could merge these changes with the latest master, I can pull this. It looks good.

**Comment by jacereda** [comment]
@paf31 I was referring to things like Just/Nothing, but in any case I found the way to include those and at the same simplified the code.

The split in different files didn't help with the merge, I hope nothing is broken...

**Comment by paf31** [comment]
It's a little hard to review this right now due to all the duplicated code. Could you please try to rebase this? Thanks.

**Pull request closed without merging**